packages/contracts-periphery/contracts/universal/op-nft/Optimist.sol

```diff
@@ -0,0 +1,160 @@
+    ) Semver(0, 0, 1) {
+        ATTESTOR = _attestor;
+        ATTESTATION_STATION = _attestationStation;
+        initialize(_name, _symbol);
+    }
+
+    /**
+     * @notice  Initialize the Optimist contract.
+     * @dev     Initializes the Optimist contract with the given parameters.
+     * @param   _name  The token name.
+     * @param   _symbol  The token symbol.
+     */
+    function initialize(string memory _name, string memory _symbol) public initializer {
+        __ERC721_init(_name, _symbol);
+        __ERC721Burnable_init();
+    }
+
+    /**
+     * @notice  Mint the Optimist token.
+     * @dev     Mints the Optimist token to the give recipient address.
+     *          Limits the number of tokens that can be minted to one per address.
+     *          The tokenId is the uint256 of the recipient address.
+     * @param   _recipient  The address of the token recipient.
+     */
+    function mint(address _recipient) public {
+        require(isWhitelisted(_recipient), "Optimist: address is not whitelisted");
+        _safeMint(_recipient, tokenIdOfOwner(_recipient));
+    }
+
+    /**
+     * @notice  Returns decimal tokenid for a given address
+     * @return  uint256 decimal tokenId
+     */
+    function baseURI() public view returns (string memory) {
+        return
+            string(
+                abi.encodePacked(
+                    ATTESTATION_STATION.attestations(
+                        ATTESTOR,
+                        address(this),
+                        bytes32("optimist.base-uri")
+                    )
+                )
+            );
+    }
+
+    /**
+     * @notice Returns the URI for the token metadata.
+     * @dev The token URI will be stored at baseURI + '/' + tokenId + .json
+     * @param tokenId The token ID to query.
+     * @return The URI for the given token ID.
+     */
+    function tokenURI(uint256 tokenId) public view virtual override returns (string memory) {
+        return
+            string(
+                abi.encodePacked(
+                    baseURI(),
+                    "/",
+                    // convert tokenId to hex string formatted like an address (20)
+                    Strings.toHexString(tokenId, 20),
+                    ".json"
+                )
+            );
+    }
+
+    /**
+     * @notice  Returns whether an address is whitelisted
+     * @dev     The whitelist is an attestation by the admin of this contract
+     * @return  boolean  Whether the address is whitelisted
+     */
+    function isWhitelisted(address _recipient) public view returns (bool) {
+        return
+            ATTESTATION_STATION
+                .attestations(ATTESTOR, _recipient, bytes32("optimist.can-mint"))
+                .length > 0;
+    }
+
+    /**
+     * @notice  Returns decimal tokenid for a given address
+     * @return  uint256 decimal tokenId
+     */
+    function tokenIdOfOwner(address _owner) public pure returns (uint256) {
```

Comment:
Since you don't have to be an owner, should this be called tokenIdOfAddress?

Reply:
Yes